src/main/java/com/minecolonies/coremod/commands/citizencommands/RespawnCitizenCommand.java

```diff
@@ -48,15 +49,16 @@ public String getCommandUsage(@NotNull final ICommandSender sender)
     public void executeSpecializedCode(@NotNull final MinecraftServer server, final ICommandSender sender, final Colony colony, final int citizenId)
     {
         final CitizenData citizenData = colony.getCitizenManager().getCitizen(citizenId);
-        final EntityCitizen entityCitizen = citizenData.getCitizenEntity();
-        sender.sendMessage(new TextComponentString(String.format(CITIZEN_DESCRIPTION, citizenData.getId(), citizenData.getName())));
+        final Optional<EntityCitizen> optionalEntityCitizen = citizenData.getCitizenEntity();
 
-        if (entityCitizen == null)
+        sender.sendMessage(new TextComponentString(String.format(CITIZEN_DESCRIPTION, citizenData.getId(), citizenData.getName())));
+        if (!optionalEntityCitizen.isPresent())
         {
-            colony.getCitizenManager().spawnCitizen(citizenData, colony.getWorld());
-            return;
+            citizenData.updateCitizenEntityIfNecessary();
         }
 
+        final EntityCitizen entityCitizen = citizenData.getCitizenEntity().get();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call "Optional#isPresent()" before accessing the value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3655)